sky/tools/gen_snapshots.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
```

Comment:
How is this different from https://github.com/flutter/engine/blob/main/sky/tools/create_macos_gen_snapshots.py ?

Reply:
The other script relies on x64, arm64 and armv7 to be build in the same machine which is not always true if running from gn + ninja context. This new script processes a single architecture at a time.